transcom/mymove
# B 20972 tio ppm

## [B-20972](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20972)

### [INT PR](https://github.com/transcom/mymove/pull/13918)

## Main-PR Specific Information
This PR was delayed for merging to main due to being downstream in implementation to [B-21393](https://github.com/transcom/mymove/pull/14335). As a result, there had to be an additional commit to resolve merge conflicts between main and this branch. The commit, [e2b4e77](https://github.com/transcom/mymove/pull/14217/commits/e2b4e77a8d0aa5d54a9b1ddba29fd7e454aef09f), only alters src/components/Office/PPM/ReviewDocumentsSidePanel/ReviewDocumentsSidePanel.jsx. Please check the commit-specific changes for this file, and ensure that none of the commit changes relative to this branch reflect as changes between this branch and main. Once that is checked, we can be sure that the merge conflicts didn't meaningfully alter the changes from the branch relative to main. Please correct me if I'm wrong on this.

## Summary

The main purpose of this ticket is to give TIOs the ability to do a read-only review of a PPM Shipment's documents and moving information when reviewing a shipment's weights. 

Currently, when a TIO navigates to the Payment Requests page, and selects "Review shipment weights" under the Billable weight section, PPMs do not display any information whatsoever and are filtered out of the shipment list. This task prevents them from being filtered out, adjusts a query that retrieves PPM Shipment information + weight tickets to also retrieve pro gear weight tickets and moving expenses, compiles all of the documents per ppm shipment (rather than document set), and displays a read-only version of the Service Counselor's overview of a PPM Shipment's documents when they review shipment weights.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### How to test

1. Access the application as a customer and prepare to make a move. 
2. At different stages of the following process, periodically log in as a TIO, navigate to the move's Payment Requests tab, select Review shipment weights, and thumb through the shipments and documents to make sure everything looks correct with the data you have created.
3. Create a move with multiple shipments. At least one should be PPM, and at least one should be non-PPM. It's helpful to make multiple PPMs to see what they'll look like at different stages in the moving process.
4. We need to generate expenses for our PPM and non-PPM shipments. For both, just follow the normal process:
5. Approve shipments as SC,
6. for non-PPM, approve as TOO,
7. for non-PPM, create service items and payment requests as prime,
8. for PPM, as a customer upload your documents, create weight tickets, pro gear weight tickets, and moving expenses.
9. If you haven't logged in to check the documents as a TIO by now, do so.
10. Approve all service items/payment requests and complete a PPM closeout for one of your PPMs.
11. Log back in as a TIO, review the shipment weights again, ensure everything is correct.

### Frontend


## Screenshots
Canvas alert is for document viewer showing document. Disabled alerts are for SC editable fields the TIO cannot access.
<img width="1540" alt="image" src="https://github.com/user-attachments/assets/09817b74-9e0c-49e1-b745-0a190b6511f6">

Only color contrast error is on the same part of the page as it always is, the divider in the top left of the header. No new images were added with text to be read.
<img width="1540" alt="image" src="https://github.com/user-attachments/assets/66701a63-0c8a-4c17-9773-c9537454d405">
